pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -631,13 +630,22 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 		m.allocatedDevices[resource] = sets.NewString()
 	}
 
+	defer func() {
+		if err == nil {
+			if m.allocatedDevices[resource] == nil {
+				m.allocatedDevices[resource] = sets.NewString()
```

Comment:
I think I'm missing something here. Which flows may possible set this to `nil`?
Once `allocatedDevices[resource]` is initialized, is never reset to nil explicitely and it's explicitely assigned only to the return value of `podDevices.devices()`, which can never be nil.

I for myself I like updating allocation once everything completed, but I'm wondering if this approach plays nice with the overall design (see comment at line 834-845). Perhaps we should review that part, but this is a much larger endeavour.